### PR TITLE
server: support dumpPartitionStats param for statsHandler

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/37977

Problem Summary:

### What is changed and how it works?

Introduce `dumpPartitionStats` for statsHandler (default true), if it was false, only global stats will be dumped of partition tables in dynamic mode.

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/dump.go [hunk 1]
```diff
@@ -119,17 +120,21 @@ func dumpJSONCol(hist *statistics.Histogram, CMSketch *statistics.CMSketch, topn
...
 	}
-	return h.DumpStatsToJSONBySnapshot(dbName, tableInfo, snapshot)
+	return h.DumpStatsToJSONBySnapshot(dbName, tableInfo, snapshot, dumpPartitionStats)
 }
 
 // DumpStatsToJSONBySnapshot dumps statistic to json.
-func (h *Handle) DumpStatsToJSONBySnapshot(dbName string, tableInfo *model.TableInfo, snapshot uint64) (*JSONTable, error) {
+func (h *Handle) DumpStatsToJSONBySnapshot(dbName string, tableInfo *model.TableInfo, snapshot uint64, dumpPartitionStats bool) (*JSONTable, error) {
+	h.mu.Lock()
+	isDynamicMode := variable.PartitionPruneMode(h.mu.ctx.GetSessionVars().PartitionPruneMode.Load()) == variable.Dynamic
```
Comment (xuyifangreeneyes) [inline statistics/handle/dump.go:136]: We can use `(*Handle).CurrentPruneMode()` here.

File: server/statistics_handler_test.go [hunk 2]
```diff
@@ -119,6 +120,33 @@ func TestDumpStatsAPI(t *testing.T) {
...
+	check := func(dumpStats bool) {
+		expectedLen := 1
+		if dumpStats {
+			expectedLen = 2
+		}
+		url := fmt.Sprintf("/stats/dump/test/test2?dumpPartitionStats=%v", dumpStats)
+		resp0, err := client.fetchStatus(url)
+		require.NoError(t, err)
+		defer func() {
+			resp0.Body.Close()
```
Comment (xuyifangreeneyes) [inline server/statistics_handler_test.go:138]: Do we need to check close error here?

**Review by xuyifangreeneyes — APPROVED** [review]
rest LGTM

**Review by qw4990 — APPROVED** [review]

**Pull request merged**